utils/tinystr/src/ascii.rs

```diff
@@ -131,16 +182,59 @@ impl<const N: usize> TinyAsciiStr<N> {
         })
     }
 
-    // TODO: This function shadows the FromStr trait. Rename?
     #[inline]
-    pub const fn from_str(s: &str) -> Result<Self, TinyStrError> {
-        Self::from_bytes_inner(s.as_bytes(), 0, s.len(), false)
+    pub(crate) const fn try_from_utf16_inner(
+        code_points: &[u16],
```

Comment:
Thought/Suggestion (optional): You don't really need the UTF-16 version to be `const` so you could implement this to instead work from `Iterator<char>` which would be convenient for other string sources like UTF-32 (python) or reader streams.

Reply:
I'm open to that in the future. I did a quick test and switching the `code_points` to take `Iterator<Item = u8>` cost ~5%. It's acceptable but I'd prefer to leave it for when we have a use case.